app/uk/gov/hmrc/eusubsidycompliancefrontend/connectors/EscConnector.scala

```diff
@@ -162,8 +133,8 @@ class EscConnectorImpl @Inject()(http: HttpClient,
         List(
           NonHmrcSubsidy(
             subsidyUsageTransactionID = journey.existingTransactionId,
-            allocationDate = currentDate,
-            submissionDate= currentDate,
+            allocationDate = journey.claimDate.value.get.toLocalDate,
```

Comment:
I am not a big fan of `.get`, can we do `getOrElse` here?

Reply:
Me neither. I guess for now we can do the exception thing, although in the long term I want to revisit it since I tend to prefer returning a suitable value e.g Either[Error, Thing] and packaging any errors up in a Left.

Reply:
yeah , even i think we should use EitherT[Future, Error, A] in our all repos which just says Future[A]. but that we can so later as a part of mass cleanup :) till now we can do getOrElse i think

Reply:
Yeah, done something like that in the past with IO so that would work quite nicely. I might add a ticket to the backlog to cover this.

Reply:
I think we just need a better way of validating/checking a journey is complete. Maybe somewhere we should validate it and convert it into a `CompleteSubidyJourney` where the fields arent' optional? Just thinking out loud. But it comes back to sensible valdiation and it being in the correct state.

Reply:
Yeah, there's definitely work to do here.

Reply:
Fix pushed.